### PR TITLE
Pass clang-format on various cpp/header files

### PR DESCRIPTION
**Pull request by nerzhul**
I ran clang-format on all my header files and choose an interesting compliant subset. Please review and we merge, those headers are not heavily modified in PR.

### CODE REVIEW

**Comment by nerzhul** [comment]
Please tell me if some files are not good and why, and i will remove them from the changeset permitting others to merge

**Comment by nerzhul** [comment]
I passed it on various cpp trivial files to fix too

File: src/chat.h [hunk 1]
```diff
@@ -38,17 +38,13 @@ struct ChatLine
...
+	    : age(0.0), name(a_name), text(a_text)
 	{
 	}
 
-	ChatLine(EnrichedString a_name, EnrichedString a_text):
-		age(0.0),
-		name(a_name),
-		text(a_text)
+	ChatLine(EnrichedString a_name, EnrichedString a_text)
+	    : age(0.0), name(a_name), text(a_text)
```
Comment (SmallJoker) [inline src/chat.h]: Spaces? Nope.

File: src/cavegen.h [hunk 2]
```diff
@@ -150,18 +154,16 @@ class CavesRandomWalk {
...
-	CavesRandomWalk(INodeDefManager *ndef,
-		GenerateNotifier *gennotify = NULL,
-		s32 seed = 0,
-		int water_level = 1,
-		content_t water_source = CONTENT_IGNORE,
-		content_t lava_source = CONTENT_IGNORE);
+	CavesRandomWalk(INodeDefManager *ndef, GenerateNotifier *gennotify = NULL,
+			s32 seed = 0, int water_level = 1,
+			content_t water_source = CONTENT_IGNORE,
+			content_t lava_source = CONTENT_IGNORE);
```
Comment (SmallJoker) [inline src/cavegen.h]: 1-tab is fine according to our code style guidelines.

**Comment by SmallJoker** [comment]
The other code looks good. Feels like this is going to be a long way until everything's styled correctly.

**Comment by nerzhul** [comment]
@SmallJoker yeah i ran clang-format on all files, 60k modifications if i remember. I remove chat.h from this changeset

**Comment by nerzhul** [comment]
i squashed the commits to have a full build with the current changeset

**Comment by nerzhul** [comment]
@SmallJoker i rebased & squashed, please tell me if it's okay for you, i will merge when build was successful

File: src/httpfetch.h [hunk 3]
```diff
@@ -126,8 +125,6 @@ void httpfetch_caller_free(unsigned long caller);
 
 // Performs a synchronous HTTP request. This blocks and therefore should
 // only be used from background threads.
-void httpfetch_sync(const HTTPFetchRequest &fetch_request,
-		HTTPFetchResult &fetch_result);
-
+void httpfetch_sync(const HTTPFetchRequest &fetch_request, HTTPFetchResult &fetch_result);
```
Comment (SmallJoker) [inline src/httpfetch.h]: Length: 91, over the limits.

Comment (nerzhul) [inline src/httpfetch.h]: strange to see how clang-format count, because he is limited to 90 chars

Comment (nerzhul) [inline src/httpfetch.h]: using python interpreter it's good

```
>>> len("void httpfetch_sync(const HTTPFetchRequest &fetch_request, HTTPFetchResult &fetch_result);")
90
```

Comment (SmallJoker) [inline src/httpfetch.h]: nvm, looks good.

Comment (SmallJoker) [inline src/httpfetch.h]: 90 is the absolute maximum. Whatever, looks good.

**Comment by nerzhul** [comment]
@sfan5 can you look at this please ?

File: src/cavegen.h (hunk 2 above)
Comment (sfan5) [inline src/cavegen.h]: 2-tab too?
The problem here is that we have to tell the formatter one of them...

File: src/drawscene.h [hunk 4]
```diff
@@ -25,15 +25,13 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
 
-void draw_scene(video::IVideoDriver *driver, scene::ISceneManager *smgr,
-		Camera &camera, Client &client, LocalPlayer *player,
-		Hud &hud, Minimap &mapper, gui::IGUIEnvironment *guienv,
-		const v2u32 &screensize, const video::SColor &skycolor,
-		bool show_hud, bool show_minimap);
+void draw_scene(video::IVideoDriver *driver, scene::ISceneManager *smgr, Camera &camera,
+		Client &client, LocalPlayer *player, Hud &hud, Minimap &mapper,
+		gui::IGUIEnvironment *guienv, const v2u32 &screensize,
+		const video::SColor &skycolor, bool show_hud, bool show_minimap);
```
Comment (sfan5) [inline src/drawscene.h]: while it may be useful to compact this, i would prefer if it didn't do this...

File: src/event_manager.h [hunk 5]
```diff
@@ -24,36 +24,34 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <list>
 #include <map>
 
-class EventManager: public MtEventManager
+class EventManager : public MtEventManager
 {
 	static void receiverReceive(MtEvent *e, void *data)
 	{
-		MtEventReceiver *r = (MtEventReceiver*)data;
+		MtEventReceiver *r = (MtEventReceiver *)data;
```
Comment (sfan5) [inline src/event_manager.h]: why did it add a space here?

File: src/keycode.h [hunk 6]
```diff
@@ -34,16 +34,16 @@ class KeyPress
 	KeyPress();
 	KeyPress(const char *name);
 
-	KeyPress(const irr::SEvent::SKeyInput &in, bool prefer_character=false);
+	KeyPress(const irr::SEvent::SKeyInput &in, bool prefer_character = false);
```
Comment (sfan5) [inline src/keycode.h]: is this part of our guidelines?

File: src/unittest/test_nodedef.cpp [hunk 7]
```diff
@@ -55,7 +56,7 @@ void TestNodeDef::testContentFeaturesSerialization()
 
 	std::ostringstream os(std::ios::binary);
 	f.serialize(os, LATEST_PROTOCOL_VERSION);
-	//verbosestream<<"Test ContentFeatures size: "<<os.str().size()<<std::endl;
+	// verbosestream<<"Test ContentFeatures size: "<<os.str().size()<<std::endl;
```
Comment (sfan5) [inline src/unittest/test_nodedef.cpp]: I understand why its doing this but this is quite stupid nonetheless

File: src/event_manager.h (hunk 5 above)
Comment (nerzhul) [inline src/event_manager.h]: because the rule for pointers is to have * at the right, then casts are affected

File: src/keycode.h (hunk 6 above)
Comment (nerzhul) [inline src/keycode.h]: no, it's a choice of LLVM to have readable code when using equal, and i'm not afraid, i prefer having it it permit to view more easily default values

File: src/unittest/test_nodedef.cpp (hunk 7 above)
Comment (nerzhul) [inline src/unittest/test_nodedef.cpp]: it's a tool, it's not an IA, the change it just for respect the coding style

File: src/drawscene.h (hunk 4 above)
Comment (nerzhul) [inline src/drawscene.h]: both are compact, one respect 80 chars other 90 chars are you sure you want me to revert this change ?

**Comment by nerzhul** [comment]
@sfan5 waiting for your update :)

**Comment by nerzhul** [comment]
rebased and i removed some files from PR as they are not very convenient, i think i will merge it tomorrow
